wrapanapi/containers/providers/rhopenshift.py

```diff
@@ -0,0 +1,874 @@
+        entities_j = self.o_api.get('template')[1]['items']
+        for entity_j in entities_j:
+            meta = entity_j['metadata']
+            entity = Template(self, meta['name'], meta['namespace'])
+            entities.append(entity)
+        return entities
+
+    def list_docker_image(self):
+        """Returns list of images (Docker registry only)"""
+        entities = []
+        entities_j = self.o_api.get('image')[1]['items']
+        for entity_j in entities_j:
+            if 'dockerImageReference' not in entity_j:
+                continue
+            _, name, image_id, _ = Image.parse_docker_image_info(entity_j['dockerImageReference'])
+            entities.append(Image(self, name, image_id))
+        return entities
+
+    def list_deployment_config(self):
+        """Returns list of deployment configs"""
+        entities = []
+        entities_j = self.o_api.get('deploymentconfig')[1]['items']
+        for entity_j in entities_j:
+            meta = entity_j['metadata']
+            entity = DeploymentConfig(self, meta['name'], meta['namespace'])
+            entities.append(entity)
+        return entities
+
+    def deploy_template(self, template, tags=None, password='smartvm'):
+        """Deploy a VM from a template
+
+        Args:
+            template: (str) The name of the template to deploy
+            tags: (dict) dict with tags if some tag isn't passed it is set to 'latest'
+            password: this password will be set as default everywhere
+        Returns: dict with parameters necessary for appliance setup or None if deployment failed
+        """
+        self.logger.info("starting template {t} deployment".format(t=template))
+        self.does_template_exist(namespace=self.default_namespace, name=template)
+        password = password
+
+        if not self.base_url:
+            raise ValueError("base url isn't provided")
+
+        prepared_tags = {key: 'latest' for key in self.template_tags}
+        if tags:
+            not_found_tags = [tag for tag in tags.keys() if tag not in self.stream_tags]
+            if not_found_tags:
+                raise ValueError("Some passed tags {t} don't exist".format(t=not_found_tags))
+            for tag, value in tags.items():
+                prepared_tags[self.stream2template_tags_mapping[tag]] = value
+
+        # create project
+        # assuming this is cfme installation and generating project name
+        proj_id = "".join(choice(string.digits + string.lowercase) for _ in range(6))
+        proj_name = "{t}-project-{pid}".format(pid=proj_id, t=template)
+        proj_url = "{proj}.{base_url}".format(proj=proj_id, base_url=self.base_url)
+        self.logger.info("unique id {id}, project name {name}".format(id=proj_id,
+                                                                      name=proj_name))
+        self.create_project(name=proj_name)
+
+        # grant rights according to scc
+        self.logger.info("granting rights to project sa")
```

Comment:
**REQUIRED**: Please add more info to *all* the logging entries here (project name, template name, ...) so in case of debugging the logs are greppable.